### ISSUE TITLE
Hub specific configuration

### ISSUE THREAD

**Issue by michaelstein60** [ebaauw/homebridge-hue#856]
Is it possible to have unique settings for each hue hub?

**Comment by ebaauw** [comment]
What kind of settings are you looking for?

**Comment by michaelstein60** [comment]
I have non hue lights that I’d like to control via Homekit. One hue bridge could be configured with these lights and then shared through homebridge-hue. The other hubs would contain Hue lights and not share these with HomeKit. I don’t currently see a way to limit shared lights to only non-hue lights.

I also like the extended function of Hue smart buttons (long press) but would prefer if I didn’t have duplicates in homekit (one direct from Hue and one from homebridge-hue).

**Comment by ebaauw** [comment]
> I don’t currently see a way to limit shared lights to only non-hue lights.

By default, Homebridge Hue doesn’t expose Hue devices on a gen-2 Hue bridge, that natively exposes these.  Set `nativeHomeKitLights` and/or `nativeHomeKitSensors` to false to change that.  See https://github.com/ebaauw/homebridge-hue/wiki/Configuration.

> I also like the extended function of Hue smart buttons (long press) but would prefer if I didn’t have duplicates in homekit (one direct from Hue and one from homebridge-hue).

I cannot help you here.  The gen-2 Hue bridge exposes all (Friends-of-) Hue devices to HomeKit.  Remove the bridge from HomeKit if you don’t want that.

**Comment by michaelstein60** [comment]
This is why settings unique to a specific hub would help. The following example is how I envision this working:

**Hue Hub.1:**
This hub is used for Hue lights and is directly linked to HomeKit. Homebridge-hue does not share these lights with HomeKit. Hub.1 also contains a number of smart buttons and other Hue UI devices. Since these UIs don’t act as ZigBee repeaters they’re connected to this hub to take advantage of the large number of hue bulbs supplying wireless coverage.

**Hue Hub.2**
This hub is used for non-hue lights. Hub.2 is not linked to HomeKit directly, but instead shares its non-hue lights through homebridge-hue. 
————-

If hub specific configuration were possible, then the above example would configure 2 hue hubs in homebridge-hue. Hub.1 would not share lights, Hub.2 would share lights.

As an alternative, I could use some ZigBee repeaters to supply network coverage for UIs on Hub.2. If configured as above Hub.1 would be all hue lights and linked to HomeKit only. Hub.2 would contain non-hue lights and UIs. Hub.2 would not be linked to HomeKit directly. This alternative has the advantage of eliminating duplicate UI devices in homekit. I experimented with 2 hubs and found that homebridge-hue will not boot if an unlinked hue hub is found. It recursively restarts homebridge waiting for the button to be pressed on hub.

In conclusion, hub specific configuration would allow non-hue lights and other devices to be shared with HomeKit on one hub in a multi hub system without creating a bunch of duplicate lights.

The alternative of allowing homebridge to boot with unlinked hue hubs is also a workable solution with extra repeaters. This option has the advantage (for my install) of eliminating duplicate UI devices.

**Comment by ebaauw** [comment]
> If hub specific configuration were possible, then the above example would configure 2 hue hubs in homebridge-hue. Hub.1 would not share lights, Hub.2 would share lights.

Please read what I’ve written before.  If you set `lights` and leave `nativeHomeKitLights` to its default value, Homebridge Hue will expose the non-Hue lights from both bridges, i.e. none from the one with only Hue lights, and all from the other.

> The alternative of allowing homebridge to boot with unlinked hue hubs is also a workable solution with extra repeaters.

No it’s not; you will lose your HomeKit accessories if the bridge happens to be unavailable when starting Homebridge.  Set `hosts` to disable discovery and specify which bridge(s) to expose.  Again, see the Wiki.  Make sure your bridge has a fixed IP address, though.

I doubt it, but if you need more fine-grained control, create `whitelist` and/or `blacklist` resourcelinks on each bridge, to specify which resources Homebridge should expose.  See the README.
